### PR TITLE
Jetpack Boost: Refactor the connection globals

### PR DESCRIPTION
**Pull request by pyronaur**
Part of the "Refactor Global State" project (pc9hqz-286-p2).

### Other information:
- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:
* Try various connection flows to thoroughly test out the connection process and compare it to the current process. There should be no noticeable changes.

### CODE REVIEW

**Review by dilirity — DISMISSED** [review]
There's a changelog missing.

File: projects/plugins/boost/app/admin/class-config.php [hunk 1]
```diff
@@ -183,11 +182,13 @@ public static function set_getting_started( $value ) {
 	 */
 	public static function is_getting_started() {
 		// Aside from the boolean flag in the database, we also assume site already got started if they have premium features.
-		return \get_option( 'jb_get_started', false ) && ! Premium_Features::has_feature( Premium_Features::CLOUD_CSS ) && ! ( new Status() )->is_offline_mode();
```
Comment (dilirity) [inline projects/plugins/boost/app/admin/class-config.php:186]: With the removal of `jb_get_started`, I'm thinking that we don't need logic tied to it at all. Should we add a ticket to clean this up in a separate PR?

File: projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte [hunk 2]
```diff
@@ -1,100 +1,45 @@
...
-
+	async function initialize( plan: 'free' | 'premium' ) {
 		try {
 			// Make sure there is a Jetpack connection
-			await connection.initialize();
+			await initializeConnection();
 
 			// Record this selection. This must be done after the connection is initialized.
-			recordBoostEvent( 'premium_cta_from_getting_started_page_in_plugin', {} );
+			recordBoostEvent( `${ plan }_cta_from_getting_started_page_in_plugin`, {} );
```
Comment (dilirity) [inline projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte:33]: The other day I had to search for events in Boost. Having the full name of an event would be helpful in finding it.

Do you think expanding the comment to include the full names of both versions makes sense here?

**Review by dilirity — DISMISSED** [review]
Great work on simplifying this!

I left a suggestion, but other than that - it's ready to merge.

:shipit:

File: projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte (hunk 2 above)
Comment (pyronaur) [reply projects/plugins/boost/app/assets/src/js/pages/getting-started/GettingStarted.svelte:33]: Fixed

**Review by dilirity — APPROVED** [review]

**Pull request merged**